### PR TITLE
config control mechanism

### PR DESCRIPTION
**Pull request by Oratorian**
# Description

Added config control mechanism for {{SERVER_PORT}} in the startup command ./altv-server, since alt:v supports the --port startup parameter. The port can no longer be mangled with by user.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [ ] The egg was exported from the panel

### CODE REVIEW

File: gta/altv/egg-pterodactyl-alt--v.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "ALT:V": "ghcr.io/parkervcp/games:altv"
     },
     "file_denylist": [],
-    "startup": "./altv-server",
+    "startup": "./altv-server --port {{SERVER_PORT}}",
```
Comment (gOOvER) [inline gta/altv/egg-pterodactyl-alt--v.json:16]: its allready in config parsing? Why is that needed?

`"port:": "port: {{server.build.default.port}}",`

**Review by gOOvER — CHANGES_REQUESTED** [review]
and btw: egg was not exported from the panel:
L4ylnfXuD3

**Comment by Oratorian** [comment]
Sorry it wasn't exported i didn't mean to click that.

And using `startup": "./altv-server --port {{SERVER_PORT}}",` ensures that the server always starts with the allocated port in pterodactyl. Ff it is only config parsed users can mangle with it. This way pterodactyl can ensure, the server always starts with the allocated port.

**Comment by gOOvER** [comment]
> This is needed because pterodactyl only forwards allocated ports, if port 7788 is allocated but the user changes the port in the config to 7789, the server will not be online.

If the user changes it, it is automatically replaced again. Only admins can change this port and not the users

**Comment by Oratorian** [comment]
> > This is needed because pterodactyl only forwards allocated ports, if port 7788 is allocated but the user changes the port in the config to 7789, the server will not be online.
> 
> If the user changes it, it is automatically replaced again. Only admins can change this port and not the users

No it is not as seen on the screenshots.

The user can change the port in server.toml and the pterodactyl set allocation is ignored.

image
image
image

**Comment by gOOvER** [comment]
then sth with parsing is wrong

**Comment by Oratorian** [comment]
I did some checking and this line seems only parsed during first setup, it's what is creating the initial server.toml

after that the server just reads all values from server.toml which makes users able to change the port in server.toml which would result in a server not being accessable because docker only fordwards the in panel allocated port.

That's why --port as standard parameter is usefull, this will always run with the allocated port from the panel and ignore the port value from the config.

**Comment by lilkingjr1** [comment]
It *should* be parsed/replaced every startup, so unless the server is super slow and the user replaces it before the server can read the file, it shouldn't be an issue. Regardless, I feel like this could be resolved fairly quickly by just removing the Configuration Files line for replacing the port and simply moving it to the Startup Command like you did (and exporting it from the panel, as required). In most cases for Eggs, I think parameters placed in the Startup Command are preferred over Configuration Files (if the application allows). So, if done right, I think this PR would likely be favorable.

**Comment by QuintenQVD0** [comment]
@Oratorian Wy the close, I was waiting on a revieuw but we were going to merge this

**Pull request closed without merging**